ambry-api/src/main/java/com.github.ambry/clustermap/ClusterParticipantFactory.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package com.github.ambry.clustermap;
+
+/**
+ * A factory interface to get a {@link ClusterParticipant}
+ */
+public interface ClusterParticipantFactory {
```

Comment:
reconsider this interface and the factory pattern. As discussed, the type of `ClusterParticipant` is tightly coupled to the the type of `ClusterMap` - may not make sense to have diff factories for them.